source/val/ValidationState.h

```diff
@@ -197,7 +203,11 @@ class ValidationState_t {
   spv_capability_mask_t
       module_capabilities_;  /// Module's declared capabilities.
 
-  std::unordered_map<uint32_t, Id> all_definitions_;
+  /// List of all instructions in the order they appear in the binary
+  std::list<Instruction> ordered_instructions_;
```

Comment:
Please use std::deque as it is better than std::list for this use case:  More densely packed data (fewer pointers), more cache friendly, and still has stable internal references since you only use emplace_back to add to it.  Also, emplace_back and back are const time.